src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -89,7 +55,7 @@ public T WithExtraQueryParameters(Dictionary<string, string> extraQueryParameter
         {
             CommonParameters.ExtraQueryParameters = extraQueryParameters ??
                 new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-            return (T)this;
+            return this as T;

```

Comment:
was this suggested by the compiler?

Reply:
Yes, `(T)this` fails to compile with addition of another layer of abstraction.